test/mysql-matrix/mzcompose.py

```diff
@@ -0,0 +1,74 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from materialize.mysql import (
+    retrieve_invalid_ssl_context_for_mysql,
+    retrieve_ssl_context_for_mysql,
+)
+from materialize.mzcompose.composition import Composition
+from materialize.mzcompose.services.materialized import Materialized
+from materialize.mzcompose.services.mysql import MySql
+from materialize.mzcompose.services.test_certs import TestCerts
+from materialize.mzcompose.services.testdrive import Testdrive
+
+MYSQL_VERSIONS = ["5.7.44", "8.3.0"]
+
+SERVICES = [
+    Materialized(
+        additional_system_parameter_defaults={
+            "log_filter": "mz_storage::source::mysql=trace,info"
+        },
+    ),
+    MySql(),
+    MySql(
+        name="mysql-replica",
+        additional_args=[
+            "--gtid_mode=ON",
+            "--enforce_gtid_consistency=ON",
+            "--skip-replica-start",
+            "--server-id=2",
+        ],
+    ),
+    TestCerts(),
+    Testdrive(volumes_extra=["../mysql-cdc:/workdir/mysql-cdc"], default_timeout="60s"),
+]
+
+
+def workflow_default(c: Composition) -> None:
+    for mysql_version in MYSQL_VERSIONS:
+        print(f"--- Testing MySQL {mysql_version}")
+        test_mysql_version(c, mysql_version)
+
+
+def test_mysql_version(c: Composition, mysql_version: str) -> None:
+    with c.override(MySql(version=mysql_version)):
```

Comment:
do we need to override the replica version too?

Reply:
Yeah, good point!